staging/src/k8s.io/kubectl/pkg/cmd/clusterinfo/clusterinfo_show_featuregates.go

```diff
@@ -0,0 +1,231 @@
+)
+
+func (o *ClusterInfoShowFeatureGatesOptions) Complete(restClientGetter genericclioptions.RESTClientGetter, cmd *cobra.Command) error {
+	printer, err := o.PrintFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+
+	o.PrintObj = printer.PrintObj
+
+	config, err := restClientGetter.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	o.CoreClient, err = corev1client.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	o.AppsClient, err = appsv1client.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	o.Timeout, err = cmdutil.GetPodRunningTimeoutFlag(cmd)
+	if err != nil {
+		return err
+	}
+
+	o.Namespace, _, err = restClientGetter.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.RESTClientGetter = restClientGetter
+	o.LogsForObject = polymorphichelpers.LogsForObjectFn
+
+	return nil
+}
+
+// feature gates status can be got from the apiserver metrics
+// 1. use `kubectl get --raw /metrics` to get the metrics.
+// 2. filter kubernetes_feature_enabled metrics and save it into a map, and the map key is
+// the feature gate name, the value is a struct contains name, stage and its status.
+// The number in the metrics is the status of the feature gate: 0 for enabled, 1 for disabled.
+// The metric is like `kubernetes_feature_enabled{name="ProcMountType",stage="ALPHA"} 0`
+// 3. Print them accordingly.
+// if All is true, keep the map and print them all
+// if All is false, filter those feature gates that is in stage ALPHA and disabled, and also filter
+// those feature gates that is in stage BETA and enabled, and filter all feature GA
+func (o *ClusterInfoShowFeatureGatesOptions) Run(f cmdutil.Factory) error {
+	restClient, err := f.RESTClient()
+	if err != nil {
+		return err
+	}
+	out := &bytes.Buffer{}
+	err = rawhttp.RawGet(restClient, genericclioptions.IOStreams{In: os.Stdin, Out: out, ErrOut: os.Stderr}, "/metrics")
+	if err != nil {
+		return err
+	}
+
+	fgMap := map[string]fgStatus{}
+	scanner := bufio.NewScanner(out)
+	for scanner.Scan() {
+		line := scanner.Text()
+		fgName, stageStr, status, err := parseLine(line)
+		if err != nil {
+			continue
+		}
+		fgMap[fgName] = fgStatus{
+			name:    fgName,
+			stage:   Stage(stageStr),
+			enabled: status == 1,
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		return err
+	}
+
+	// Output map
+	if o.All {
+		for _, fg := range fgMap {
+			o.printGateStatus(fg)
+		}
+	} else {
+		for _, fg := range fgMap {
+			if fg.stage == "GA" ||
+				(fg.stage == "ALPHA" && !fg.enabled) ||
+				(fg.stage == "BETA" && fg.enabled) {
+				continue
+			}
+			o.printGateStatus(fg)
+		}
+	}
+
+	return nil
+}
+
+func (o *ClusterInfoShowFeatureGatesOptions) printGateStatus(fg fgStatus) {
+	fmt.Fprintf(o.Out, "%s=%v (%s)\n", fg.name, fg.enabled, fg.stage)
+}
+
+type Stage string
+
+type fgStatus struct {
+	name    string
+	stage   Stage
+	enabled bool
+}
+
+func parseLine(line string) (string, Stage, int, error) {
```

Comment:
I think it would be better to just use the prometheus parser library to do this directly, rather than string matching: https://pkg.go.dev/github.com/prometheus/common@v0.42.0/expfmt#TextParser